src/java/picard/sam/markduplicates/MarkDuplicates.java

```diff
@@ -521,16 +587,12 @@ private void generateDuplicateIndexes(final boolean useBarcodes, final boolean i
         // First just do the pairs
         log.info("Traversing read pair information and detecting duplicates.");
         for (final ReadEndsForMarkDuplicates next : this.pairSort) {
-            if (firstOfNextChunk == null) {
```

Comment:
Can you set `firstOfNextChunk = null` prior to line 610 (`for (final ReadEndsForMarkDuplicates next : this.fragSort)`).  I am trying to debate why `firstOfNextChunk` wouldn't need to be reset to `null` before starting on the fragments.